react/features/prejoin/middleware.web.ts

```diff
@@ -78,6 +82,25 @@ MiddlewareRegistry.register(store => next => async action => {
     return next(action);
 });
 
+/**
+ * Set up state change listener to show a prompt when the display name
+ * is required, but missing.
+ */
+StateListenerRegistry.register(
```

Comment:
Didn't we say we'd do this when clicking the Join button instead of this? Sorry, I know we went back and forth...

Reply:
Yes, that's for mobile, but on the web it works ok, so I thought we'd live it like that, wdyt?

Reply:
Does it make things much more complex to put things together? I think it would makle the logic a bit more tight.